code/game/objects/items/cigs_lighters.dm

```diff
@@ -279,16 +279,19 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 			return
 	smoketime -= delta_time
 	if(smoketime <= 0)
-		new type_butt(location)
-		if(ismob(loc))
-			to_chat(M, "<span class='notice'>Your [name] goes out.</span>")
-		qdel(src)
-		return
+		go_out(location, loc)
 	open_flame()
 	if((reagents?.total_volume) && (nextdragtime <= world.time))
 		nextdragtime = world.time + dragtime SECONDS
 		handle_reagents()
 
+/obj/item/clothing/mask/cigarette/proc/go_out(turf/location, mob/living/M) //handles cigs putting themselves out after being emptied
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
Also, proper autodoc.